pkg/apis/gcpfirewall/v1alpha1/types.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Protocol defines network protocols supported for GCP firewall.
+type Protocol string
+
+// CIDR defines a IP block.
+// TODO(b/186120065) Modify the validation to include IPv6 CIDRs with FW 3.0 support.
```

Comment:
Remove internal bug references, replace this with your github id

Reply:
Done.